Scripts/SAPSetupAssistantValidationScripts/SetupAssistantSAPValidateKerberosConfiguration.ps1

```diff
@@ -0,0 +1,275 @@
+    The OPDG Service Principal
+#>
+
+param (
+    [string]$sapCclDllPath,
+    [string]$sapCclIniPath,
+    [string]$sapServicePrincipalName,
+    [string]$servicePrincipal,
+    [string]$opdgServicePrincipal
+)
+
+# Required for Kerberos checks
+$module = Get-Module -ListAvailable -Name ActiveDirectory
+
+if ($module) {
+    Write-Host -NoNewLine "{'step': 'CheckActiveDirectoryPowerShellModule', 'status': 'Success', 'message': 'The ActiveDirectory PowerShell module is installed.'},"
+}
+else {
+    Write-Host -NoNewLine  "{'step': 'CheckActiveDirectoryPowerShellModule', 'status': 'Error', 'message': 'The ActiveDirectory PowerShell module is not installed. Kerberos checks can not be performed. Install it by following this link: https://go.microsoft.com/fwlink/?linkid=2240709.'},"
+}
+
+#CheckCommonCryptoLibVersion
+$path = $sapCclDllPath
+$minimumVersion = "8.5.25.0"
+
+try {
+    $actualVersion = [System.Diagnostics.FileVersionInfo]::GetVersionInfo($path).FileVersion
+}
+catch {
+    Write-Host -NoNewLine "{'step': 'CheckCommonCryptoLibVersion', 'status': 'Error', 'message': 'CommonCryptoLibVersion error occurred: $_. Did you specify the correct path to the SAP CommonCryptoLib DLL?'},"
+    return
+}
+
+if ([version]$actualVersion -ge [version]$minimumVersion) {
+    Write-Host -NoNewLine "{'step': 'CheckCommonCryptoLibVersion', 'status': 'Success', 'message': 'SAP Common Crypto Library is running version $actualVersion.'},"
+}
+else {
+    Write-Host -NoNewLine "{'step': 'CheckCommonCryptoLibVersion', 'status': 'Error', 'message': 'SAP Common Crypto Library is running an old version $actualVersion. Download the latest version of the SAP Common Crypto Library from SAP.'},"
+}
+
+#CheckSapCryptoIniFile
+$path = $sapCclIniPath
+
+if (-not $path) {
+    Write-Host -NoNewLine "{'step': 'CheckSapCryptoIniFile', 'status': 'Warning', 'message': 'Skipping SapCryptoIniFile check because no file path was provided.'},"
+    return
+}
+
+$clientRole = "ccl/snc/enable_kerberos_in_client_role=1"
+
+try {
+    $iniContent = Get-Content -Path $path -ErrorAction Stop
+}
+catch {
+    Write-Host -NoNewLine "{'step': 'CheckSapCryptoIniFile', 'status': 'Error', 'message': 'SapCryptoIniFile error occurred: $_. Did you specify the correct path to the sapcrypto.ini file?'},"
+    return
+}
+
+if ($iniContent -match [regex]::Escape($clientRole).Replace('=', '\s*=\s*')) {
+    Write-Host -NoNewLine "{'step': 'CheckSapCryptoIniFile', 'status': 'Success', 'message': 'sapcrypto.ini has the correct parameters: $clientRole.'},"
+}
+else {
+    Write-Host -NoNewLine "{'step': 'CheckSapCryptoIniFile', 'status': 'Error', 'message': 'sapcrypto.ini is missing the correct parameters. Modify sapcrypto.ini and add $clientRole.'},"
+}
+
+#checkSystemEnvVariable -variableName "CCL_PROFILE"
+
+if ($env:CCL_PROFILE) {
+    Write-Host -NoNewLine "{'step': 'CheckSapCryptoIniFile', 'status': 'Success', 'message': 'System Environment variable 'CCL_PROFILE' exists'},"
```

Comment:
The message string is enclosed in single quotes, but the variable in the string is also enclosed in single quotes - might want to escape those

Reply:
Updated